### PR TITLE
Adds nullability specifiers for Core Animation and Foundation classes

### PR DESCRIPTION
**Pull request by dfowjtm**

### CODE REVIEW

File: Foundation/Error Utilities/TWTErrorUtilities.h [hunk 1]
```diff
@@ -32,6 +32,7 @@
 
 #import <Foundation/Foundation.h>
 
+
```
Comment (prachigauriar) [inline Foundation/Error Utilities/TWTErrorUtilities.h]: Everything in this file should probably be `_Nonnull`.

File: Foundation/KVO/TWTKeyValueObserver.h [hunk 2]
```diff
@@ -38,7 +41,7 @@ typedef void(^TWTKeyValueObserverChangeBlock)(id observedObject, NSDictionary *c
  */
 @interface TWTKeyValueObserver : NSObject
 
-@property (nonatomic, weak, readonly) id object;
+@property (nonatomic, weak, readonly, nullable) id object;
```
Comment (prachigauriar) [inline Foundation/KVO/TWTKeyValueObserver.h]: Everything below this should also probably be nonnull.

File: Foundation/Subclass Responsibility/NSException+TWTSubclassResponsibility.h [hunk 3]
```diff
@@ -39,7 +39,7 @@
  @param selector The selector to which the receiver will be responding. This is typically _cmd.
  @result A new NSInternalInconsistencyException
  */
-+ (instancetype)twt_subclassResponsibilityExceptionWithReceiver:(id)receiver selector:(SEL)selector;
++ (nullable instancetype)twt_subclassResponsibilityExceptionWithReceiver:(id _Nullable)receiver selector:(SEL _Nullable)selector;
```
Comment (prachigauriar) [inline Foundation/Subclass Responsibility/NSException+TWTSubclassResponsibility.h]: Non-null for these too.

File: Foundation/Tree Node/TWTTreeNode.h [hunk 4]
```diff
@@ -26,17 +26,22 @@
...
 
-@property (nonatomic, strong) id item;
+@property (nonatomic, strong, nullable) id item;
 @property (nonatomic, assign, getter = isExpanded) BOOL expanded;
-@property (nonatomic, copy) NSArray *children; // TWTTreeNode
+@property (nonatomic, copy, null_resettable) NSArray<TWTTreeNode *> *children;
 
-@property (nonatomic, readonly, weak) TWTTreeNode *parent;
-@property (nonatomic, readonly, strong) NSIndexPath *indexPath;
+@property (nonatomic, readonly, weak, nullable) TWTTreeNode *parent;
```
Comment (prachigauriar) [inline Foundation/Tree Node/TWTTreeNode.h]: Don’t need to specify nullable on weak properties. They’re inherently nullable.

File: Foundation/Error Utilities/TWTErrorUtilities.h (hunk 1 above)
Comment (dfowjtm) [inline Foundation/Error Utilities/TWTErrorUtilities.h]: Agreed

File: Foundation/KVO/TWTKeyValueObserver.h (hunk 2 above)
Comment (dfowjtm) [inline Foundation/KVO/TWTKeyValueObserver.h]: Including instances of `target` parameters? Will those get sent up the responder chain?

Comment (prachigauriar) [inline Foundation/KVO/TWTKeyValueObserver.h]: Nope.

Comment (prachigauriar) [inline Foundation/KVO/TWTKeyValueObserver.h]: KVO doesn’t use the responder chain. target shouldn’t be nullable.

Comment (dfowjtm) [inline Foundation/KVO/TWTKeyValueObserver.h]: nonnilified

**Comment by dfowjtm** [comment]
side note: is this going to require a major version update? These changes could introduce compiler errors

**Comment by prachigauriar** [comment]
All versions of Toast are major version updates, because we never hit 1.0.

**Comment by prachigauriar** [comment]
:+1:

**Pull request merged**